lib/symengine/basic.rb

```diff
@@ -11,8 +11,13 @@ def free_symbols
     end
     
     def abs
-        return SymEngine::abs(self)
+      SymEngine::abs(self)
+    end
+    def to_proc()
+      if self.free_symbols.count > 1
+        raise ArgumentError, "Too many free symbols! Only 1 allowed. Found #{self.free_symbols.count}."
```

Comment:
Can you explain the difficulty with the free symbols order?
FYI, In SymEngine we keep `pi*r**2` as an ordered dictionary. `{pi : 1, r : 2}`. This order is not dependent on which symbol was used first in the formula, `pi*r**2`